core/base/src/utils/encoding.ts

```diff
@@ -41,3 +43,17 @@ export const toUint8Array = (value: string | bigint): Uint8Array =>
 
 export const fromUint8Array = (value: Uint8Array): string =>
   (new TextDecoder()).decode(value);
+
+export const equals = (lhs: Uint8Array, rhs: Uint8Array): boolean =>
+  lhs.length === rhs.length && lhs.every((v, i) => v === rhs[i]);
+
+export const concat = (...args: Uint8Array[]): Uint8Array => {
+  const length = args.reduce((acc, curr) => acc + curr.length, 0);
+  const result = new Uint8Array(length);
+  let offset = 0;
+  args.forEach((arg) => {
+    result.set(arg, offset);
+    offset += arg.length;
+  });
+  return result;
+}
```

Comment:
should we namespace the u8array stuff as well?

Reply:
I think so. I don't think we want to have a free-standing `equals` and `concat` lying around.